domain/schema/model.go

```diff
@@ -482,8 +482,39 @@ ON unit (unit_id);
 CREATE INDEX idx_unit_application
 ON unit (application_uuid);
 
-CREATE UNIQUE INDEX idx_unit_net_node
+CREATE INDEX idx_unit_net_node
 ON unit (net_node_uuid);
+
+CREATE TABLE unit_state (
+    unit_uuid       TEXT PRIMARY KEY,
+    uniter_state    TEXT,
+    storage_state   TEXT,
+    secret_state    TEXT,
+    CONSTRAINT      fk_unit_state_unit
+        FOREIGN KEY (unit_uuid)
+        REFERENCES  unit(uuid)
+);
+
+CREATE TABLE unit_state_charm (
+    unit_uuid       TEXT,
+    key             TEXT,
+    value           TEXT NOT NULL,
+    PRIMARY KEY     (unit_uuid, key),
+    CONSTRAINT      fk_unit_state_charm_unit
+        FOREIGN KEY (unit_uuid)
+        REFERENCES  unit(uuid)
+);
+
+CREATE TABLE unit_state_relation (
+    unit_uuid       TEXT,
+    key             TEXT,
+    value           TEXT NOT NULL,
+    PRIMARY KEY     (unit_uuid, key),
+    CONSTRAINT      fk_unit_state_relation_unit
+        FOREIGN KEY (unit_uuid)
+        REFERENCES  unit(uuid)
+);
```

Comment:
Can we add a note to these, that they're essentially local charm state persisted to the db.